perf-tool/src/monitor.cpp

```diff
@@ -55,7 +57,19 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
         sprintf(str, "0x%x", hash);
         j["monitorHash"] = str;
     }
-
+   
+    /*monitor for protecting the hashtable*/
+    std::mutex monitorMapMutex;
```

Comment:
The mutex needs to be static, otherwise every thread will use a new monitor and there is no synchronization.